### PR TITLE
DAT 3464

### PR DESCRIPTION
**Pull request by wwillard7800**

### CODE REVIEW

**Review by nvoxland — APPROVED** [review]

File: liquibase-core/src/main/java/liquibase/integration/commandline/Main.java [hunk 1]
```diff
@@ -191,6 +197,20 @@ public static int run(String[] args) throws LiquibaseException {
...
             }
 
+            //
+            // Look for characters which cannot be handled
+            //
+            for (int i=0; i < args.length; i++) {
+                int index = checkArg(args[i]);
+                if (index >= 0) {
+                    String message =
+                        "Invalid, non-printable character detected in command line at position " +
```
Comment (SteveDonie) [inline liquibase-core/src/main/java/liquibase/integration/commandline/Main.java]: Should mention the code point in the message for easier debugging. "Invalid, non printable character #${index} detected in command line..."

Comment (SteveDonie) [reply liquibase-core/src/main/java/liquibase/integration/commandline/Main.java]: Oops - just realized the return value is the index where the character is found, not which character was found. Could change the logic of checkArg to return a struct with both values - what was found and where.

**Review by SteveDonie — COMMENTED** [review]
Suggestion for change to information in the message

**Review by SteveDonie — APPROVED** [review]
Nice!

**Comment by bbrown-at-datical** [comment]
@wwillard7800 ~ Can a unit test be provided to cover this code to assert that valid characters are not interfered with and invalid characters are removed as expected?

**Review by SteveDonie — APPROVED** [review]

**Comment by nvoxland** [comment]
Merged in as https://github.com/liquibase/liquibase/pull/955

**Pull request closed without merging**